src/GraphQL/Validation/Complexity/AnalysisContext.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using GraphQLParser.AST;
+using GraphQLParser.Visitors;
+
+namespace GraphQL.Validation.Complexity
+{
+    internal sealed class AnalysisContext : INodeVisitorContext
+    {
+        public double AvgImpact { get; set; }
+
+        public double CurrentSubSelectionImpact { get; set; }
+
+        public double CurrentEndNodeImpact { get; set; }
+
+        public FragmentComplexity CurrentFragmentComplexity { get; set; } = null!;
+
+        public ComplexityResult Result { get; } = new ComplexityResult();
+
+        public int LoopCounter { get; set; }
+
+        public int MaxRecursionCount { get; set; }
+
+        public bool FragmentMapAlreadyBuilt { get; set; }
+
+        public Dictionary<string, FragmentComplexity> FragmentMap { get; } = new Dictionary<string, FragmentComplexity>();
+
+        public CancellationToken CancellationToken => default;
+
+        public void AssertRecursion()
+        {
+            if (LoopCounter++ > MaxRecursionCount)
+            {
+                throw new InvalidOperationException("Query is too complex to validate.");
+            }
+        }
+
+        public static double? GetImpactFromArgs(GraphQLField node) //TODO: variables support
```

Comment:
Do you know if the prior code supported variables or not?

Reply:
99% sure that not

Reply:
Code in master checks for literal value only.

Reply:
Really glad you are taking a look the complexity analyzer. 🥇

Reply:
100% sure - code checks for `IntValue` (literal) node only. It knows nothing about `VariableReference` node.

Reply:
I am also glad. I was lazy to explore it.